doc/api/fs.md

```diff
@@ -1524,11 +1524,27 @@ If `options` is a string, then it specifies the encoding. Example:
 ```js
 fs.readFile('/etc/passwd', 'utf8', callback);
 ```
+*Note*: When the path is a directory, the behavior of
+`fs.readFile()` and [`fs.readFileSync()`][] is platform-specific. On macOS,
+Linux, and Windows, an error will be returned. On FreeBSD, the contents
+of the files in the directory will be returned.
```

Comment:
@adityaanandmc No problem
```
$ node -p "fs.readFileSync('test', 'utf-8')"

```
The same as
```js
Buffer.from([3, 0, 0, 0]).toString("utf-8")
````
What did you expect ? :smiley:
Yes, macOS works as expected(throws an error), but the subject is FreeBSD, and "test" is a directory.
> On FreeBSD, the contents of the files in the directory will be returned.

This is wrong.
Linux(Darwin too) doesn't allow using the read syscall to read bytes from a directory, but FreeBSD does :neutral_face:
And the result doesn't look like the contents of the files in the directory